src/integration-test/java/com/commercetools/sync/integration/producttypes/ProductTypeSyncIT.java

```diff
@@ -0,0 +1,827 @@
+
+        final ProductTypeSync productTypeSync = new ProductTypeSync(spyProductTypeSyncOptions);
+
+        final ProductTypeSyncStatistics productTypeSyncStatistics = productTypeSync
+            .sync(singletonList(newProductTypeDraft))
+            .toCompletableFuture().join();
+
+        // Since the error message and exception is coming from commercetools, we don't test the actual message and
+        // exception
+        verify(spyProductTypeSyncOptions).applyErrorCallback(any(), any());
+
+        AssertionsForStatistics.assertThat(productTypeSyncStatistics).hasValues(1, 0, 0, 1);
+    }
+
+    @Test
+    public void sync_WithoutAttributeType_ShouldExecuteCallbackOnErrorAndIncreaseFailedCounter() {
+        final AttributeDefinitionDraft attributeDefinitionDraft = AttributeDefinitionDraftBuilder
+            .of(
+                null,
+                "attr_name_1",
+                LocalizedString.ofEnglish("attr_label_1"),
+                true
+            )
+            .attributeConstraint(AttributeConstraint.NONE)
+            .inputTip(LocalizedString.ofEnglish("inputTip1"))
+            .inputHint(TextInputHint.SINGLE_LINE)
+            .isSearchable(false)
+            .build();
+
+        final ProductTypeDraft newProductTypeDraft = ProductTypeDraft.ofAttributeDefinitionDrafts(
+            PRODUCT_TYPE_KEY_1,
+            null,
+            PRODUCT_TYPE_DESCRIPTION_1,
+            singletonList(attributeDefinitionDraft)
+        );
+
+        final ProductTypeSyncOptions productTypeSyncOptions = ProductTypeSyncOptionsBuilder
+            .of(CTP_TARGET_CLIENT)
+            .build();
+
+        ProductTypeSyncOptions spyProductTypeSyncOptions = spy(productTypeSyncOptions);
+
+        final ProductTypeSync productTypeSync = new ProductTypeSync(spyProductTypeSyncOptions);
+
+        final ProductTypeSyncStatistics productTypeSyncStatistics = productTypeSync
+            .sync(singletonList(newProductTypeDraft))
+            .toCompletableFuture().join();
+
+        // Since the error message and exception is coming from commercetools, we don't test the actual message and
+        // exception
+        verify(spyProductTypeSyncOptions).applyErrorCallback(any(), any());
+
+        AssertionsForStatistics.assertThat(productTypeSyncStatistics).hasValues(1, 0, 0, 1);
+    }
+
+    @Test
+    public void sync_WithSeveralBatches_ShouldReturnProperStatistics() {
+        // Default batch size is 50 (check ProductTypeSyncOptionsBuilder) so we have 2 batches of 50
+        final List<ProductTypeDraft> productTypeDrafts = IntStream
+            .range(0, 100)
+            .mapToObj(i -> ProductTypeDraft.ofAttributeDefinitionDrafts(
+                "product_type_key_" + Integer.toString(i),
+                "product_type_name_" + Integer.toString(i),
+                "product_type_description_" + Integer.toString(i),
+                Arrays.asList(ATTRIBUTE_DEFINITION_DRAFT_1)
```

Comment:
you can use here instead: `singletonList(ATTRIBUTE_DEFINITION_DRAFT_1)`

Reply:
601ffb1e1af54363e7d5a52b771054f5d7d1a900